src/components/infinite-data-table.jsx

```diff
@@ -0,0 +1,427 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  CellMeasurer,
+  CellMeasurerCache,
+  MultiGrid,
+  AutoSizer,
+  InfiniteLoader,
+} from 'react-virtualized';
+import { serializableDeepAreEqual } from '../utils';
+import '../styles/components/data-table.scss';
+
+const DEFAULT_WIDTH = 200;
+const DEFAULT_HEIGHT = 30;
+const MIN_HEIGHT = 30;
+const NUMBER_COLUMN = {
+  label: '#',
+  numberColumn: true,
+  width: 40,
+};
+const SELECT_COLUMN = {
+  selectColumn: true,
+  width: 40,
+};
+
+export const InfiniteDataTablePreprocessor = (props) => {
+  // This functional component prepends the data and columns being passed into InfiniteDataTableCore
+  // based on consumer choices (ie showRowNumbers, selectable, showHeader).
+  const {
+    showRowNumbers, selectable, showHeader, data: inData, columns: inColumns,
+  } = props;
+  const data = showHeader ? [{}, ...inData] : [...inData];
+  let columns = [...inColumns];
+  if (showRowNumbers) {
+    columns = [NUMBER_COLUMN, ...columns];
+  }
+  if (selectable) {
+    columns = [SELECT_COLUMN, ...columns];
+  }
+  return <InfiniteDataTableCore {...props} data={data} columns={columns} />;
+};
+
+export class InfiniteDataTableCore extends Component {
+  cache = new CellMeasurerCache({
+    fixedWidth: true,
+    DEFAULT_HEIGHT,
+    MIN_HEIGHT,
+  });
+
+  constructor(props) {
+    super(props);
+    this.getColumnWidth = this.getColumnWidth.bind(this);
+    this.cellRenderer = this.cellRenderer.bind(this);
+    this.isRowLoaded = this.isRowLoaded.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    const { data, columns } = this.props;
+    const { data: prevData } = prevProps;
+    if (!serializableDeepAreEqual(data, prevData)) {
+      for (let rowIndex = prevData.length; rowIndex <= data.length + 1; rowIndex += 1) {
+        for (let columnIndex = 0; columnIndex < columns.length; columnIndex += 1) {
+          this.cache.clear(rowIndex, columnIndex);
+        }
+      }
+    }
+  }
+
+  getHeader() {
+    const { columns } = this.props;
+    return columns.map(column => ({ label: column.label || '', render: x => x }));
```

Comment:
I'm not sure what `x` is. Is there a more explicit name?

Reply:
`getHeader` is a relic from a previous life. Removed the method.